### PR TITLE
Add `$lockMode` parameter to `EntityManager::refresh()`

### PR DESCRIPTION
**Pull request by michnovka**
Cleanup of 3.0.x branch for https://github.com/doctrine/orm/pull/10040

### CODE REVIEW

File: lib/Doctrine/ORM/EntityManagerInterface.php [hunk 1]
```diff
@@ -133,6 +132,22 @@ public function createQueryBuilder(): QueryBuilder;
      */
     public function find(string $className, mixed $id, LockMode|int|null $lockMode = null, int|null $lockVersion = null): object|null;
 
+    /**
+     * Refreshes the persistent state of an object from the database,
+     * overriding any local changes that have not yet been persisted.
+     *
+     * @param object            $object   The object to refresh.
```
Comment (derrabus) [inline lib/Doctrine/ORM/EntityManagerInterface.php]:
```suggestion
```

File: tests/Doctrine/Performance/Mock/NonProxyLoadingEntityManager.php [hunk 2]
```diff
@@ -201,9 +201,9 @@ public function detach(object $object): void
         $this->realEntityManager->detach($object);
     }
 
-    public function refresh(object $object): void
+    public function refresh(object $object, LockMode|int|null $lockMode = null): void
     {
-        $this->realEntityManager->refresh($object);
+        $this->realEntityManager->refresh($object, $lockMode);
     }
```
Comment (derrabus) [inline tests/Doctrine/Performance/Mock/NonProxyLoadingEntityManager.php:207]: Backport to 2.14.x?

File: UPGRADE.md [hunk 3]
```diff
@@ -508,6 +508,11 @@ following classes and methods:
 
 Use `toIterable()` instead.
 
+## BC BREAK: Changed `EntityManagerInterface#refresh($entity)`, `EntityManagerDecorator#refresh($entity)` and `UnitOfWork#refresh($entity)` signatures
+
+The new signatures of these methods add an optional `LockMode|int|null $lockMode`
+param with default `null` value (no lock).
+
```
Comment (derrabus) [inline UPGRADE.md]: We add new entries at the top of the file.

File: tests/Doctrine/Performance/Mock/NonProxyLoadingEntityManager.php (hunk 2 above)
Comment (michnovka) [reply tests/Doctrine/Performance/Mock/NonProxyLoadingEntityManager.php:207]: yes, https://github.com/doctrine/orm/pull/10197

**Review by derrabus — APPROVED** [review]

**Pull request merged**